File: SignalR Server/Models/SignalRMessage.cs
namespace TAPP_SignalR.Models
{
   public enum SignalRClientMethods
   {
      reset,
      join,
      attack
   }

   public enum SignalRServerMethods
   {
      error,
      info,
      reset,
      joined,
      attack,
      attackResponse,
      getBoardState
   }

   public enum PlayerTypes
   {
      player1,
      player2,
      spectator
   }

   public class SignalRClientMessage
   {
      public SignalRClientMethods Method { get; set; }
      public dynamic? Message { get; set; }
   }

   public class SignalRServerMessage
   {
      public SignalRServerMethods Method { get; set; }
      public dynamic? Message { get; set; }

      public SignalRServerMessage(SignalRServerMethods method, dynamic message)
      {
         Method = method;
         Message = message;
      }
   }
}

File: TAPP_SignalR_Sheru/src/app/signalr.message.ts
import { Coord } from "./game/GameVariables"

export enum SignalRClientMethods {
   reset,
   join,
   attack
}

export enum SignalRServerMethods {
   error,
   info,
   reset,
   joined,
   attack,
   attackResponse,
   getBoardState
}

export enum PlayerTypes {
   player1,
   player2,
   spectator
}

export interface MessageJoin {
   type: PlayerTypes
}

export interface MessageAttack {
   coord: Coord
}

export interface SignalRMessage {
   Method: SignalRClientMethods,
   Message: any
}

export interface SignalRServerMessage {
   method: SignalRServerMethods,
   message: any
}